firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -122,6 +136,61 @@ private void createMutationQueue() {
             + "PRIMARY KEY (uid, path, batch_id))");
   }
 
+  private void removeAcknowledgedMutations() {
+    SQLitePersistence.Query mutationQueuesQuery =
+        new SQLitePersistence.Query(
+            db, "SELECT uid, last_acknowledged_batch_id FROM mutation_queues");
+
+    mutationQueuesQuery.forEach(
+        mutationQueueEntry -> {
+          String uid = mutationQueueEntry.getString(0);
+          long lastAcknowledgedBatchId = mutationQueueEntry.getLong(1);
+
+          SQLitePersistence.Query mutationsQuery =
+              new SQLitePersistence.Query(
+                      db, "SELECT mutations FROM mutations WHERE uid = ? AND batch_id <= ?")
+                  .binding(uid, lastAcknowledgedBatchId);
+          mutationsQuery.forEach(
+              value -> {
+                try {
+                  MutationBatch batch =
+                      serializer.decodeMutationBatch(
+                          com.google.firebase.firestore.proto.WriteBatch.parseFrom(
+                              value.getBlob(0)));
+                  removeMutationBatch(uid, batch);
+                } catch (InvalidProtocolBufferException e) {
+                  throw fail("MutationBatch failed to parse: %s", e);
+                }
+              });
+        });
+  }
+
+  private void removeMutationBatch(String uid, MutationBatch batch) {
+    int batchId = batch.getBatchId();
+
+    SQLiteStatement mutationDeleter =
+        db.compileStatement("DELETE FROM mutations WHERE uid = ? AND batch_id = ?");
+    mutationDeleter.bindString(1, uid);
+    mutationDeleter.bindLong(2, batchId);
+    int deleted = mutationDeleter.executeUpdateDelete();
+    hardAssert(deleted != 0, "Mutation batch (%s, %d) did not exist", uid, batch.getBatchId());
+
+    SQLiteStatement indexDeleter =
+        db.compileStatement(
+            "DELETE FROM document_mutations WHERE uid = ? AND path = ? AND batch_id = ?");
```

Comment:
Do we need the path? Don't we want to delete all rows from a batch? Could we do `DELETE FROM document_mutations WHERE uid = ? AND batch_id = ?` and then maybe not even need the serializer?